### PR TITLE
Use TRFLSQFitter instead of LevMarLSQFitter

### LINKED ISSUE #3189: Stop using LevMarLSQFitter as recommended by astropy
_Issue [JDAT-4776](https://jira.stsci.edu/browse/JDAT-4776) was created on JIRA by [Pey-Lian Lim](https://jira.stsci.edu/secure/ViewProfile.jspa?name=lim):_

Search for usage of LevMarLSQFitter and replace it with something better. See astropy/astropy#16983 for more details.

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix #3189

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-4776)

### CODE REVIEW

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py [hunk 1]
```diff
@@ -876,7 +876,7 @@ def calculate_photometry(self, dataset=None, aperture=None, background=None,
 
                 # Fit Gaussian1D to radial profile data.
                 if self.fit_radial_profile:
-                    fitter = LevMarLSQFitter()
+                    fitter = TRFLSQFitter()
```
Comment (pllim) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:879]: @larrybradley , is this replacement reasonable or another one is better? Also see https://github.com/astropy/photutils/issues/1895

Comment (larrybradley) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:879]: That should work fine.  I don't think you are using parameter bounds, so `LMLSQFitter` is also an option.

Comment (pllim) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:879]: Thanks! Well, not yet. No idea if that is going to be future request or not. 😬

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:879]: is there any consequence on performance between the two for this use-case?

Comment (pllim) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:879]: Nothing that we have to worry about. I am using what Astropy now suggests in their docs. Tom R made that change upstream and he has been also working on modeling performance there, so he wouldn't recommend anything that would make things worse.

**Review by kecnry — APPROVED** [review]

**Comment by pllim** [comment]
Thanks for the review!

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py (hunk 1 above)
Comment (larrybradley) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:879]: In my testing, `TRFLSQFitter` is about 4-5% slower than `LevMarLSQFitter`.  However, this difference will not be noticeable here.

**Pull request merged**